vslib/src/SwitchStateBase.cpp

```diff
@@ -2424,3 +2439,191 @@ void SwitchStateBase::debugSetStats(
+
+    m_system_port_list.clear();
+
+    for (uint32_t i = 0; i < sys_port_count; i++)
+    {
+        SWSS_LOG_DEBUG("create system port index %u", i);
+
+        sai_object_id_t system_port_id;
+
+        CHECK_STATUS(create(SAI_OBJECT_TYPE_SYSTEM_PORT, &system_port_id, m_switch_id, 0, NULL));
+        m_system_port_list.push_back(system_port_id);
+
+        sai_attribute_t attr;
+
+        attr.id = SAI_SYSTEM_PORT_ATTR_CONFIG_INFO;
+        attr.value.sysportconfig = sys_port_cfg_list[i];
+        CHECK_STATUS(set(SAI_OBJECT_TYPE_SYSTEM_PORT, system_port_id, &attr));
+
+        attr.id = SAI_SYSTEM_PORT_ATTR_TYPE;
+        attr.value.s32 = SAI_SYSTEM_PORT_TYPE_REMOTE;
+        if ((int32_t)sys_port_cfg_list[i].attached_switch_id == voq_switch_id)
+        {
+            attr.value.s32 = SAI_SYSTEM_PORT_TYPE_LOCAL;
+
+            //This is system port of local port. Set the oid of the local port corresponding to this system port
+            auto map = m_switchConfig->m_corePortIndexMap;
+            if (map)
+            {
+                auto& corePortIndexVector = map->getCorePortIndexVector();
+                size_t n_map = corePortIndexVector.size();
+                size_t idx;
+                for (idx = 0; idx < n_map; idx++)
+                {
+                    if (corePortIndexVector[idx][0] == sys_port_cfg_list[i].attached_core_index &&
+                            corePortIndexVector[idx][1] == sys_port_cfg_list[i].attached_core_port_index &&
+                            idx < m_port_list.size())
+                    {
+                        sai_attribute_t lp_attr;
+                        lp_attr.id = SAI_SYSTEM_PORT_ATTR_PORT;
+                        //m_port_list entries are in the same order as lane maps. The core port index maps are in the
+                        //same order as the lane maps. So m_port_list at the index corresponding to the core port index map
+                        //will be the port corresponding to the system port with core port index matching core port index map
+                        lp_attr.value.oid = m_port_list.at(idx);
+                        CHECK_STATUS(set(SAI_OBJECT_TYPE_SYSTEM_PORT, system_port_id, &lp_attr));
+                        break;
+                    }
+                }
+                if (idx >= n_map)
+                {
+                    SWSS_LOG_ERROR("Core port index not found for system port %d, System port's local port oid is not set!",
+                            sys_port_cfg_list[i].port_id, sai_serialize_object_id(m_switch_id).c_str());
+                }
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Core port index map for switch %s is NULL, Local port oid is not set for system port %d!",
+                        sai_serialize_object_id(m_switch_id).c_str(), sys_port_cfg_list[i].port_id);
+            }
+        }
+        CHECK_STATUS(set(SAI_OBJECT_TYPE_SYSTEM_PORT, system_port_id, &attr));
+    }
+
```

Comment:
empty line befor each if/for and after }

Reply:
Fixed in PR #701 and PR #703